PORTING.md

```diff
@@ -0,0 +1,128 @@
+## Envoy Porting
```

Comment:
Should we cover cross-architecture porting here? E.g. ARM, PowerPC, etc.

Reply:
Absolutely. A 32-bit android PR reiterated this point today. We should likely break this down into subject areas that aren't as much "windows specific" as "architecture variances", e.g. 32 bit android or arm or linux, vs P64 Windows, vs ILP64 linux. Each of these should have short breakouts, but the suggestions writ large should focus on the individual commonly misunderstood subjects and how they should be coded to remain portable.